product_pricelist_direct_print/wizards/product_pricelist_print.py

```diff
@@ -66,6 +66,7 @@ class ProductPricelistPrint(models.TransientModel):
         " the last X ordered products will be obtained for the report."
     )
     summary = fields.Text()
+    print_child_categories = fields.Boolean()
```

Comment:
```suggestion
    print_child_categories = fields.Boolean(default=True)
```

I guess, if we select category / categories we expect childs categories to be selected IMO.

Reply:
yes, but I wrote 

```
@api.onchange("categ_ids")
    def _onchange_categ_ids(self):
        self.print_child_categories = len(self.categ_ids) > 0
```
        
so, when we select category / categories, print_child_categories will be True first

Reply:
and, If print_child_categories=True first. When we don't select category / categories, we get wrong domain in
```
if self.print_child_categories:
            domain = expression.AND(
                [domain, [("categ_id", "child_of", self.categ_ids.ids)]]
            )
elif self.categ_ids:
            domain = expression.AND([domain, [("categ_id", "in", self.categ_ids.ids)]])
 ```